client/components/jetpack/sidebar/style.scss

```diff
@@ -2,6 +2,22 @@
 
 .site-selector__jetpack-cloud {
 	display: block;
+
+	.site-selector__actions {
+		padding: 16px;
+	}
+
+	.site-selector__add-new-site {
+		border-color: var( --color-sidebar-text );
+		border-radius: 2px;
+
+		&,
+		&:hover {
+			box-shadow: 1px solid var( --color-neutral-20 );
+			background-color: var( --color-sidebar-background );
+			color: var( --color-sidebar-text );
+		}
```

Comment:
Why are we removing the hover effect on Jetpack?
This is how it's looking:
![no-hover](https://user-images.githubusercontent.com/779993/188488667-55267c9a-c80c-4ae0-beb8-d7b5d27ca2c2.gif)

I suggest moving the CSS only to the base class so the hover will look different:
```suggestion
		box-shadow: 1px solid var( --color-neutral-20 );
		background-color: var( --color-sidebar-background );
		color: var( --color-sidebar-text );
```
![hover-effect](https://user-images.githubusercontent.com/779993/188488662-2f892377-b981-4f38-9dcd-dccb239d5d05.gif)

Reply:
Using the calypso and jetpack cloud link on the PR, the styling look different:

**Calypso**
![image](https://user-images.githubusercontent.com/47489215/188495626-b6321137-bb4f-408a-a71c-6bbd38f5665e.png)

**Jetpack Cloud**
![image](https://user-images.githubusercontent.com/47489215/188496187-3e79d337-f1e0-4329-9807-727643be8bc1.png)

Is the color dependent on some global styles?

Reply:
@sejas I still see the hover there:

![Screen Capture on 2022-09-06 at 12-53-48](https://user-images.githubusercontent.com/727413/188617698-909d92d5-78ef-4ccf-8a2b-8efafea28590.gif)

What is the URL that you are using?

Reply:
@wojtekn The URL I was using is the Jetpack Dashboard for agencies: https://cloud.jetpack.com/dashboard

It seems the new PR doesn't touch the Jetpack css file: `client/components/jetpack/sidebar/style.scss`, and now the button is looking good.
https://github.com/Automattic/wp-calypso/pull/67459/files

![jetpack-add-new-site-button](https://user-images.githubusercontent.com/779993/188840034-4abeee6e-22e7-4a55-b518-fca90985bb2f.gif)